gradoop-flink/src/test/java/org/gradoop/flink/algorithms/gelly/trianglecounting/GellyTriangleCountingTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.trianglecounting;
+
+import org.gradoop.common.GradoopTestUtils;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test class for {@link GellyTriangleCounting}
+ */
+public class GellyTriangleCountingTest extends GradoopFlinkTestBase {
+
+  /**
+   * Test triangle counting for cases:
+   * <pre>
+   *   - no edges
+   *   - 1 triplet
+   *   - 2 triangles in a directed graph
+   *   - 2 triangles in an undirected graph
+   *   - 8 triangles in directed social graph from {@link GradoopTestUtils#SOCIAL_NETWORK_GDL_FILE}
+   * </pre>
+   */
+  @Test
+  public void testTriangleCounting() throws Exception {
+
+    String graphString =
+      "noEdges[" +
+      "/* 2 not connected vertices */" +
+      "(v0 {id:0, value:\"A\"})" +
```

Comment:
Vertices with the same variable has not to be defined multiple times in different log graphs. Maybe you define the vertices one outside the logical graphs and put only edges inside the logical graphs. Also - is its better (but not necessary) to define unique edge variables for unique edges. E.g. you use e1 multiple times for different edges.